app/javascript/components/widgets/guests/carousel.jsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Query } from 'react-apollo';
+import gql from 'graphql-tag';
+import { sortBy, prop, append, lensPath, set } from 'ramda';
+import styled from 'styled-components';
+import icon from '../../../../assets/images/icons/icon-calendar.svg';
+import { parseTime, isDateToday } from '../../../lib/datetime';
+import { LoadingMessage, DisconnectedMessage } from '../messages/message';
+
+const subscribeAnnouncementPublished = gql`
+  subscription onAnnouncementPublished {
+    announcementPublished {
+      message
+      people
+      company
+      publishOn
+      id
+    }
+  }
+`;
+
+const getPrimaryLocationAnnouncements = gql`
+  {
+    primaryLocation {
+      dayAnnouncements {
+        message
+        people
+        company
+        publishOn
+        id
+      }
+    }
+  }
+`;
+
+const AnnouncementTime = styled.div``;
+
+const AnnouncementTitle = styled.div``;
+
+const AnnouncementWrapper = styled.div`
+  display: flex;
+  height: 5vh;
+  padding: 1vh;
+`;
+
+const AnnouncementIcon = styled.div`
+  background: url(${icon});
+  width: 3vw;
+  background-repeat: no-repeat;
+`;
+
+class SubscribedAnnouncements extends React.Component {
+  componentDidMount() {
+    this.props.subscribeToPublishedAnnouncements();
+  }
+
+  render() {
+    const { dayAnnouncements } = this.props;
+
+    return (
+      <div>
+        {sortBy(prop('publishOn'), dayAnnouncements).map(item => (
+          <AnnouncementWrapper key={item.id}>
+            <AnnouncementIcon />
+            <div>
+              <AnnouncementTime>{parseTime(item.publishOn)}</AnnouncementTime>
+              <AnnouncementTitle>{item.people}</AnnouncementTitle>
+            </div>
+          </AnnouncementWrapper>
+        ))}
+      </div>
+    );
+  }
+}
+SubscribedAnnouncements.propTypes = {
+  dayAnnouncements: PropTypes.shape({
+    message: PropTypes.string.isRequired,
+    people: PropTypes.string.isRequired,
+    company: PropTypes.string.isRequired,
+    publishOn: PropTypes.string.isRequired,
+    id: PropTypes.string.isRequired,
+  }).isRequired,
+  subscribeToPublishedAnnouncements: PropTypes.func.isRequired,
+};
+
+const Guests = () => (
```

Comment:
can we update this component name to match the file name? also, i think it should be named something like `GuestsCarouselItem` until we come up with a better name.

Reply:
I created a ticket to switch the names from carousel to tab. Might be better to remain consistent for now and switch every thing in one go.

Reply:
oh ok then.